backend/src/main/java/gov/cdc/usds/simplereport/api/export/DataHubUploadController.java

```diff
@@ -0,0 +1,49 @@
+package gov.cdc.usds.simplereport.api.export;
+
+import gov.cdc.usds.simplereport.service.DataHubUploaderService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+
+/**
+ * Created by tomn on 11Dec2020
+ */
+@RestController
+@RequestMapping(value = "/export")
+@Validated
+public class DataHubUploadController {
+
+    private final DataHubUploaderService _hubuploadservice;
+
+    public DataHubUploadController(DataHubUploaderService us) {
+        this._hubuploadservice = us;
+    }
+
+    @GetMapping(value = "/uploadTestEvent")
+    public String uploadTestEventCSVToDataHub(@RequestParam String apikey,
+                                              @RequestParam(defaultValue = "") String startupdateby
+    ) throws IOException {
+        return _hubuploadservice.uploadTestEventCVSToDataHub(apikey, startupdateby);
+    }
+
+    @GetMapping(value = "/testEvent", produces = {"text/csv"})
+    public ResponseEntity<?> exportTestEventCSV(HttpServletResponse response,
+                                                @RequestParam(defaultValue = "") String startupdateby) throws IOException {
+        response.setContentType("text/csv");
+        DateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
+        String currentDateTime = dateFormatter.format(new Date());
+
+        String headerKey = "Content-Disposition";
+        String headerValue = "attachment; filename=testEvents_" + currentDateTime + ".csv";
+        response.setHeader(headerKey, headerValue);
+        response.getWriter().print(_hubuploadservice.creatTestCVSForDataHub(startupdateby));
```

Comment:
This is not a great way of handling this, though it won't bite us too badly for this data size—this will double-buffer the output string, and I think we can avoid at least one of those.

Reply:
Yeah. The save-to-file was existing code that I just moved. Not sure how much it will be used in the future. The whole process of taking our data out of a DB, reducing it's fidelity to a CSV, then sending that as a post-body seems like a near-term only solution?

As you point out, streaming/double-buffer is clearly the better option here. With the very restrictive limit of 999 rows the Data-Hub has now, it shouldn't matter too much. But if we ever want to support a Facility downloading all their own data, then it will need to be fixed.

Re: ConfigurationProperties holding all the data. Putting the MAX_ROW in there is a good idea, I'll do that. I wasn't sure about the secret API keys (one for dev, another production). We should have a quick meeting to discuss so I understand all the issued.

Re: persisting state in the DB is the next step. I'm just trying to get done in chunks to make it more manageable to code (as per Tim). Also, schema changes probably need a bigger discussion.

Reply:
Since I didn't modify this code for this change, I'm going to defer it a later date.
I created an issue #391 so we don't forget.